sdk/pinocchio/src/sysvars/tests/slot_hashes_tests.rs

```diff
@@ -0,0 +1,855 @@
+                for digit_ref in b58_digits_rev.iter_mut() {
+                    let temp_val = ((*digit_ref as u32) << 8) | carry;
+                    *digit_ref = (temp_val % 58) as u8;
+                    carry = temp_val / 58;
+                }
+                while carry > 0 {
+                    b58_digits_rev.push((carry % 58) as u8);
+                    carry /= 58;
+                }
+            }
+            for &byte_val in input_bytes {
+                if byte_val == 0 {
+                    b58_digits_rev.push(0)
+                } else {
+                    break;
+                }
+            }
+            let mut output_chars = Vec::with_capacity(b58_digits_rev.len());
+            for &digit_val in b58_digits_rev.iter().rev() {
+                output_chars.push(BASE_58[digit_val as usize]);
+            }
+            assert_eq!(expected_b58.as_bytes(), output_chars.as_slice());
+        }
+        check_base58(
+            &SLOTHASHES_ID,
+            "SysvarS1otHashes111111111111111111111111111",
+        );
+    }
+
+    fn create_mock_data(entries: &[(u64, [u8; 32])]) -> Vec<u8> {
+        let num_entries = entries.len() as u64;
+        let data_len = NUM_ENTRIES_SIZE + entries.len() * ENTRY_SIZE;
+        let mut data = std::vec![0u8; data_len];
+        data[0..NUM_ENTRIES_SIZE].copy_from_slice(&num_entries.to_le_bytes());
+        let mut offset = NUM_ENTRIES_SIZE;
+        for (slot, hash) in entries {
+            data[offset..offset + SLOT_SIZE].copy_from_slice(&slot.to_le_bytes());
+            data[offset + SLOT_SIZE..offset + ENTRY_SIZE].copy_from_slice(hash);
+            offset += ENTRY_SIZE;
+        }
+        data
+    }
+
+    fn generate_mock_entries(
+        num_entries: usize,
+        start_slot: u64,
+        strategy: DecrementStrategy,
+    ) -> Vec<(u64, [u8; 32])> {
+        let mut entries = Vec::with_capacity(num_entries);
+        let mut current_slot = start_slot;
+        for i in 0..num_entries {
+            let hash_byte = (i % 256) as u8;
+            let hash = [hash_byte; 32];
+            entries.push((current_slot, hash));
+            let random_val = simple_prng(i as u64);
+            let decrement = match strategy {
+                DecrementStrategy::Strictly1 => 1,
+                DecrementStrategy::Average1_05 => {
+                    if random_val % 20 == 0 {
+                        2
+                    } else {
+                        1
+                    }
+                }
+                #[allow(dead_code)] // May be used by benchmarks
+                DecrementStrategy::Average2 => {
+                    if random_val % 2 == 0 {
+                        1
+                    } else {
+                        3
+                    }
+                }
+            };
+            current_slot = current_slot.saturating_sub(decrement);
+        }
+        entries
+    }
+
+    #[cfg(feature = "std")]
+    mod std_tests {
+        use super::*;
```

Comment:
*minor*

I think it is better to keep all the tests at the same level, and just mark those that require `std` as such individually.

The extra module name shows in the test name when you run `cargo test`.
And increases the indentation level.
But, the fact that the test only makes sense when `std` is fully supported like more of an implementation detail of a particular test.
For example, maybe when an API changes, a test that used to be a `core` test now need `std` or vice versa.